test/Unit/PowerShell/PowerShellManagerTests.cs

```diff
@@ -50,132 +77,175 @@ internal static PowerShellManager NewTestPowerShellManager(ConsoleLogger logger,
             return pwsh != null ? new PowerShellManager(logger, pwsh) : new PowerShellManager(logger, id: 2);
         }
 
-        internal static AzFunctionInfo NewAzFunctionInfo(string scriptFile, string entryPoint)
+        private static (AzFunctionInfo, PowerShellManager) PrepareFunction(string scriptFile, string entryPoint)
         {
-            RpcFunctionMetadata.ScriptFile = scriptFile;
-            RpcFunctionMetadata.EntryPoint = entryPoint;
-            RpcFunctionMetadata.Directory = Path.GetDirectoryName(scriptFile);
-            return new AzFunctionInfo(RpcFunctionMetadata);
+            s_functionLoadRequest.Metadata.ScriptFile = scriptFile;
+            s_functionLoadRequest.Metadata.EntryPoint = entryPoint;
+            s_functionLoadRequest.Metadata.Directory = Path.GetDirectoryName(scriptFile);
+
+            FunctionLoader.LoadFunction(s_functionLoadRequest);
+            var funcInfo = FunctionLoader.GetFunctionInfo(s_functionLoadRequest.FunctionId);
+            var psManager = NewTestPowerShellManager(s_testLogger);
+
+            return (funcInfo, psManager);
         }
 
-        // Helper method to wait for debugger to attach and set a breakpoint.
-        internal static void Break()
+        public void Dispose()
         {
-            while (!System.Diagnostics.Debugger.IsAttached)
+            FunctionLoader.ClearLoadedFunctions();
+            s_testLogger.FullLog.Clear();
+        }
+
+        [Fact]
+        public void InvokeBasicFunctionWorks()
```

Comment:
I feel like these Facts could be refactored into a Theory with multiple test cases since it's mostly just a chance in the ps1 that gets processed and the expected result.

I won't block on this though.

Reply:
Sound good. Issue #222 was opened to track this.